db/changes_view.go

```diff
@@ -35,29 +35,45 @@ func (dbc *DatabaseContext) getChangesInChannelFromView(
 	start := time.Now()
 	// Query the view:
 	optMap := changesViewOptions(channelName, endSeq, options)
-	base.LogTo("Cache", "  Querying 'channels' view for %q (start=#%d, end=#%d, limit=%d)", channelName, options.Since.SafeSequence()+1, endSeq, options.Limit)
-	vres := channelsViewResult{}
-	err := dbc.Bucket.ViewCustom(DesignDocSyncGatewayChannels, ViewChannels, optMap, &vres)
-	if err != nil {
-		base.Logf("Error from 'channels' view: %v", err)
-		return nil, err
-	} else if len(vres.Rows) == 0 {
-		base.LogTo("Cache", "    Got no rows from view for %q", channelName)
-		return nil, nil
-	}
 
-	// Convert the output to LogEntries:
-	entries := make(LogEntries, 0, len(vres.Rows))
-	for _, row := range vres.Rows {
-		entry := &LogEntry{
-			Sequence:     uint64(row.Key[1].(float64)),
-			DocID:        row.ID,
-			RevID:        row.Value.Rev,
-			Flags:        row.Value.Flags,
-			TimeReceived: time.Now(),
+	var activecount int
+	var entries LogEntries
+	
+	//Loop until we have consumed limit log entries
+	for {
+		base.LogTo("Cache", "  Querying 'channels' view for %q (start=#%d, end=#%d, limit=%d)", channelName, options.Since.SafeSequence()+1, endSeq, options.Limit)
+		vres := channelsViewResult{}
+		err := dbc.Bucket.ViewCustom(DesignDocSyncGatewayChannels, ViewChannels, optMap, &vres)
+		if err != nil {
+			base.Logf("Error from 'channels' view: %v", err)
+			return nil, err
+		} else if len(vres.Rows) == 0 {
+			base.LogTo("Cache", "    Got no rows from view for %q", channelName)
+			return nil, nil
```

Comment:
If we run out of records in the database before hitting the limit over multiple iterations, it looks like we're going to fall into this code and return an empty result (instead of whatever we gathered in our previous results).

Shouldn't this be returning `entries` if len(entries) > 0?